web/scripts/generateZonesConfig.ts

```diff
@@ -31,12 +31,14 @@ const mergeZones = () => {
   ]);
   const zones = filesWithDirectory.reduce((zones, filepath) => {
     const zoneConfig: any = yaml.load(fs.readFileSync(filepath, 'utf8'));
-    zoneConfig.bounding_box?.forEach((point: number[]) => {
-      point[0] = round(point[0], 4);
-      point[1] = round(point[1], 4);
-    });
+    if (zoneConfig?.bounding_box) {
+      for (const point of zoneConfig.bounding_box) {
+        point[0] = round(point[0], 4);
+        point[1] = round(point[1], 4);
+      }
+    }
 
-    for (const key in zoneConfig) {
```

Comment:
What's the rule for not allowing `x in y`?

Reply:
Something about duplicating the underlying prototype objects and slowing down performance. So `Object.keys(object)` creates a new iterator and then of loops over it as if it was an array.

I didn't dig too much into it though and just fixed it 😅